### PR TITLE
Draft UI is completed and able to help test connection.

### PR DESCRIPTION
**Pull request by Lotunnnnny**
Besides what I added in a new module called "JavaWeb", I have modified the .gitignore file to exclude all the .idea configuration files.
Belows are snapshots of what I make for our JavaWeb UI.
296b223aa31dc9abf3dd919cad7f5a3
93c84bdbb805befcede6b1321ff2cad
652cccdcd866f57f5f1ea260571f350

### CODE REVIEW

**Comment by EmadNaguiDarwish** [comment]
I see that all of the frontend code is vue.js, so is it better to make a separate vue.js application?

**Comment by MetallicFocus** [comment]
@EmadNaguiDarwish Do you mean like splitting the code inside vue.js file by page?

**Comment by EmadNaguiDarwish** [comment]
I mean this is a java app, we can make it a vue.js app because we are not using any java.

**Comment by Lotunnnnny** [comment]
@EmadNaguiDarwish It could be, but as far as I'm concerned, vue.js is basically all frontend thing and it has to cooperate with backend language like Java to function meaningfully, which makes it a JavaWeb application. And I haven't started the backend part.
Btw, do you mean a java applet by "java app"? 'Cuz I see you want me to write frontend codes with Java.

File: JavaWeb/web/main.html [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+					chat1: {
+						avatar: "",
+						name: "Team Zero",
+						time: "23:56",
+						content: "Nobody: Hello World!"
+					},
+					chat2: {
+						avatar: "./img/Harley.png",
+						name: "Harley",
+						time: "0:26",
```
Comment (MetallicFocus) [inline JavaWeb/web/main.html]: "00:26"

File: JavaWeb/web/sign_up.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	                name: '',
+					pwd: ''
+	              },
+	              rules: {
+					email: [
+					  { required: true, message: 'Please enter your email address.', trigger: 'blur' }
+					],
+	                name: [
+	                  { required: true, message: 'Please enter your user name.', trigger: 'blur' },
+	                  { max: 12, message: 'No more than 12 characters.', trigger: 'blur' }
```
Comment (MetallicFocus) [inline JavaWeb/web/sign_up.html]: We'll change these later, for now it's good!

**Comment by EmadNaguiDarwish** [comment]
Ok, we can leave it as it is.

**Comment by Lotunnnnny** [comment]
@MetallicFocus Thanks for looking through my codes. You are so careful that find the mistakes I overlooked. I don't know how to reply your comments on my codes, so I reply to you here.

**Comment by Lotunnnnny** [comment]
@MetallicFocus Sorry, I just got you wrong. New reply: yup, we could discuss the principles later.

**Pull request merged**